Pal/src/host/Linux-SGX/tools/ra-tls/secret_prov_attest.c

```diff
@@ -44,28 +44,39 @@ static mbedtls_ssl_context g_ssl;
 static mbedtls_pk_context g_my_ratls_key;
 static mbedtls_x509_crt g_my_ratls_cert;
 
-static uint8_t* provisioned_secret = NULL;
-static size_t provisioned_secret_size = 0;
+static uint8_t* g_provisioned_secret = NULL;
+static size_t g_provisioned_secret_size = 0;
+
+void secret_provision_free_resources(void) {
+    mbedtls_x509_crt_free(&g_my_ratls_cert);
+    mbedtls_pk_free(&g_my_ratls_key);
+    mbedtls_net_free(&g_verifier_fd);
+    mbedtls_ssl_free(&g_ssl);
+    mbedtls_ssl_config_free(&g_conf);
+    mbedtls_x509_crt_free(&g_verifier_ca_chain);
+    mbedtls_ctr_drbg_free(&g_ctr_drbg);
+    mbedtls_entropy_free(&g_entropy);
+}
 
 int secret_provision_get(uint8_t** out_secret, size_t* out_secret_size) {
     if (!out_secret || !out_secret_size)
         return -EINVAL;
 
-    *out_secret      = provisioned_secret;
-    *out_secret_size = provisioned_secret_size;
+    *out_secret      = g_provisioned_secret;
+    *out_secret_size = g_provisioned_secret_size;
     return 0;
 }
 
 void secret_provision_destroy(void) {
-    if (provisioned_secret && provisioned_secret_size)
+    if (g_provisioned_secret && g_provisioned_secret_size)
 #ifdef __STDC_LIB_EXT1__
-        memset_s(provisioned_secret, 0, provisioned_secret_size);
+        memset_s(g_provisioned_secret, 0, g_provisioned_secret_size);
 #else
-        memset(provisioned_secret, 0, provisioned_secret_size);
+        memset(g_provisioned_secret, 0, g_provisioned_secret_size);
 #endif
-    free(provisioned_secret);
-    provisioned_secret      = NULL;
-    provisioned_secret_size = 0;
+    free(g_provisioned_secret);
```

Comment:
seems need to add"if (g_provisioned_secret)", otherwise free() may lead to exception when calling secret_provision_destroy() twice